src/main/java/uk/gov/companieshouse/charges/data/service/ChargesService.java

```diff
@@ -163,32 +159,35 @@ public Optional<ChargesApi> findCharges(final String companyNumber, final Pageab
                             companyNumber));
             return Optional.empty();
         }
-        var result = companyMetrics.map(metrics -> createChargesApi(charges, metrics));
+        Optional<ChargesApi> result = Optional.ofNullable(createChargesApi(charges,
+                companyMetrics));
         logger.debug(String.format("Finished : findCharges charges found for Company Number %s ",
                 companyNumber
         ));
         return result;
     }
 
-    private ChargesApi createChargesApi(List<ChargesDocument> charges, MetricsApi metrics) {
+    private ChargesApi createChargesApi(List<ChargesDocument> charges,
+            Optional<MetricsApi> metrics) {
+        if (metrics.isEmpty()) { // No metrics so no need to create ChargesApi result.
```

Comment:
We can still populate charges data like total count, items etc if metrics is not present